src/hooks/useBestTrade.ts

```diff
@@ -56,9 +56,10 @@ export function useBestTrade(
   const [routerPreference] = useRouterPreference()
   const routingAPITrade = useRoutingAPITrade(
     tradeType,
-    autoRouterSupported && shouldGetTrade ? debouncedAmount : undefined,
+    debouncedAmount,
```

Comment:
don't pass an undefined `amountSpecified` to `useRoutingAPITrade` to force the polling interrupt - let's just have a separate variable below to control that

Reply:
I think if `amountSpecified` is undefined it should be passed immediately, otherwise you have a weird delay in the UI from debouncing.

```suggestion
    amountSpecified ? debouncedAmount : undefined,
```

Another way to specify this would be in the definition of `debouncedAmount`.